pkg/collector/corechecks/cluster/orchestrator/orchestrator_processor.go

```diff
@@ -292,8 +338,40 @@ func processNodesList(nodesList []*corev1.Node, groupID int32, cfg *config.Orche
 		})
 	}
 
+	clusterMessage := extractClusterMessage(nodeCount, kubeletVersions, apiServerVersions, podCap, podAllocatable, memoryAllocatable, memoryCap, cpuAllocatable, cpuCap, cfg, clusterID, groupID)
+	if orchestrator.SkipKubernetesResource(types.UID(clusterID), clusterMessage.Cluster.ResourceVersion, orchestrator.K8sCluster) {
+		return messages, nil, nil
+	}
+
 	log.Debugf("Collected & enriched %d out of %d nodes in %s", len(nodeMsgs), len(nodesList), time.Now().Sub(start))
-	return messages, nil
+	return messages, clusterMessage, nil
+}
+
+func extractClusterMessage(nodeCount int32, kubeletVersions map[string]int32, apiServerVersions map[string]int32, podCap uint32, podAllocatable uint32, memoryAllocatable uint64, memoryCap uint64, cpuAllocatable uint64, cpuCap uint64, cfg *config.OrchestratorConfig, clusterID string, groupID int32) *model.CollectorCluster {
+	cluster := &model.Cluster{
+		NodeCount:         nodeCount,
+		KubeletVersions:   kubeletVersions,
+		ApiServerVersions: apiServerVersions,
+		PodCapacity:       podCap,
+		PodAllocatable:    podAllocatable,
+		MemoryAllocatable: memoryAllocatable,
+		MemoryCapacity:    memoryCap,
+		CpuAllocatable:    cpuAllocatable,
+		CpuCapacity:       cpuCap,
+	}
+
+	if err := fillClusterResourceVersion(cluster); err != nil {
+		log.Warnf("Failed to compute cluster resource version: %s", err)
```

Comment:
Do we want to skip the collection of the cluster altogether if the RV cannot be computed? 
So just return err here and level above we skip this clusterMessage?